File: apps/problems/Optimal-Flight-Route/full-boilerplate/function.cpp
#include <iostream>
    #include <fstream>
    #include <vector>
    #include <string>
    #include <sstream>
    #include <climits>
    
    ##USER_CODE_HERE##
    
    int main() {
      std::ifstream file("/Users/vishal/Desktop/PROJECTS/algoearth3/apps/problems/optimal-flight route/tests/inputs/##INPUT_FILE_INDEX##.txt");
      std::vector<std::string> lines;
      std::string line;
      while (std::getline(file, line)) lines.push_back(line);
    
      file.close();
      int departureCityID;
  std::istringstream(lines[0]) >> departureCityID;
  int arrivalCityID;
  std::istringstream(lines[1]) >> arrivalCityID;
  int maxStops;
  std::istringstream(lines[2]) >> maxStops;
  int size_layoverCosts;
  std::istringstream(lines[3]) >> size_layoverCosts;
  std::vector<int> layoverCosts(size_layoverCosts);
  if(!size_layoverCosts==0) {
  	std::istringstream iss(lines[4]);
  	for (int i=0; i < size_arr; i++) iss >> arr[i];
  }
  int maxLayoverTime;
  std::istringstream(lines[4]) >> maxLayoverTime;
      list<int> result = findOptimalRoute(departureCityID, arrivalCityID, maxStops, layoverCosts, maxLayoverTime);
      std::cout << result << std::endl;
      return 0;
    }
    

File: apps/problems/Optimal-Flight-Route/full-boilerplate/function.rs
use std::fs::read_to_string;
    use std::io::{self};
    use std::str::Lines;
    
    ##USER_CODE_HERE##
    
    fn main() -> io::Result<()> {
      let input = read_to_string("/Users/vishal/Desktop/PROJECTS/algoearth3/apps/problems/optimal-flight route/tests/inputs/##INPUT_FILE_INDEX##.txt")?;
      let mut lines = input.lines();
      let departureCityID: i32 = lines.next().unwrap().parse().unwrap();
  let arrivalCityID: i32 = lines.next().unwrap().parse().unwrap();
  let maxStops: i32 = lines.next().unwrap().parse().unwrap();
  let size_layoverCosts: usize = lines.next().and_then(|line| line.parse().ok()).unwrap_or(0);
	let layoverCosts: Vec<i32> = parse_input(lines, size_layoverCosts);
  let maxLayoverTime: i32 = lines.next().unwrap().parse().unwrap();
      let result = findOptimalRoute(departureCityID, arrivalCityID, maxStops, layoverCosts, maxLayoverTime);
      println!("{}", result);
      Ok(())
    }
fn parse_input(mut input: Lines, size_arr: usize) -> Vec<i32> {
        let arr: Vec<i32> = input
            .next()
            .unwrap_or_default()
            .split_whitespace()
            .filter_map(|x| x.parse().ok())
            .collect();
    
        if size_arr == 0 {
            Vec::new()
        } else {
            arr
        }
    }
    